merlin/analysis/paralleltaskcomplete.py

```diff
@@ -0,0 +1,27 @@
+from merlin.core import analysistask
+
+
+class ParallelTaskComplete(analysistask.AnalysisTask):
+    """
+    A task to simplify snakemake construction, there is no need to explicitly
+    invoke this task in analysis files, it is inferred by the snakewriter.
+    If running outside snakemake this class is not required.
+    """
+    def __init__(self, dataSet, parameters=None, analysisName=None):
+        super().__init__(dataSet, parameters, analysisName)
+
+    def get_estimated_memory(self):
+        return 100
+
+    def get_dependencies(self):
+        # The dependency is inferred by the snakewriter
+        return [self.parameters['dependent_task']]
+
+    def get_estimated_time(self):
+        return 1
+
+    def _run_analysis(self):
+        dependentTask = self.dataSet.load_analysis_task(
+            self.parameters['dependent_task'])
+        if dependentTask.is_complete():
+            self.dataSet.record_analysis_complete(analysisTask=dependentTask)
```

Comment:
Can this go through dependentTask to create the done file? It seems like within the dependentTask function is_complete(), if the task is done, it can write the .done file itself. For example self.dataSet.record_analysis_complete(analysisTask) can be moved to line 383 of analysistask.py. Additionally, the is_complete function for parallelanalysistask can be modified so that it first checks for this file to see if it's complete rather than checking all 1000+ .done files. This should speed up this part as well. Would this work?

Reply:
This seems good to me, I've implemented it this way.